drivers/clone.c

```diff
@@ -598,22 +611,54 @@ void upsdrv_updateinfo(void)
 		return;
 	}
 
+	status_init();
+	status_set(ups.status); /* FIXME: Split token words? */
+
+	online = status_get("OL");
+
+	if (!outlet) {
+		/* Outlet was declared OFF (by upstream or assumption) */
+		status_set("OFF");
+		if (ups.load.status && !reported_off) { /* first entrance */
+			reported_off = 1;
+			upslogx(LOG_WARNING, "%s: upstream reports outlet off (setting OFF)",
+				__func__);
+		}
+	} else if (ups.load.status && reported_off) { /* first cleared */
+		reported_off = 0;
+		upslogx(LOG_INFO, "%s: upstream reports outlet no longer off (clearing OFF)",
+			__func__);
+	}
+
 	if (ups.timer.shutdown >= 0) {
 
+		status_set("FSD");
+		upslogx(LOG_WARNING, "%s: outlet shutdown initiated (setting FSD)",
+			__func__);
+
 		ups.timer.shutdown -= (suseconds_t)(difftime(now, last_poll));
 
 		if (ups.timer.shutdown < 0) {
-			const char	*val;
-
 			ups.timer.shutdown = -1;
-			outlet = 0;
 
-			val = getval("load.off");
-			if (val) {
+			if (ups.load.off) {
 				char	buf[SMALLBUF];
-				snprintf(buf, sizeof(buf), "INSTCMD %s\n", val);
+				upslogx(LOG_WARNING, "%s: sending '%s' to upstream (load.off)",
+					__func__, ups.load.off);
+				snprintf(buf, sizeof(buf), "INSTCMD %s\n", ups.load.off);
 				sstate_sendline(buf);
 			}
+
+			if (!ups.load.status) {
+				/* Better than nothing if no load.status argument was given,
```

Comment:
If no argument was given where to check the outlet's actual status, we assume it to have successfully turned off after sending `ups.load.off` for lack of a variable for checking. If an argument was given, no assumptions are made and the actual status of the outlet is checked. 

Same as before, but it was just set here and overridden later in an unexpected place `instcmd`, this way should be cleaner and more traceable with debug messages.